lib/src/Uncertainty/Distribution/PiecewiseLinearDistribution.cxx

```diff
@@ -0,0 +1,385 @@
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <cmath>
+#include "openturns/PiecewiseLinearDistribution.hxx"
+#include "openturns/RandomGenerator.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(PiecewiseLinearDistribution)
+
+static const Factory<PiecewiseLinearDistribution> Factory_PiecewiseLinearDistribution;
+
+/* Default constructor */
+PiecewiseLinearDistribution::PiecewiseLinearDistribution()
+  : ContinuousDistribution()
+  , a_(-1.0)
+  , b_(1.0)
+{
+  setName( "PiecewiseLinearDistribution" );
+  setDimension( 1 );
+  computeRange();
+}
+
+/* Parameters constructor */
+PiecewiseLinearDistribution::PiecewiseLinearDistribution(const Scalar a,
+                 const Scalar b)
+  : ContinuousDistribution()
+  , a_(a)
+  , b_(b)
+{
+  setName( "PiecewiseLinearDistribution" );
+  if (b <= a) throw InvalidArgumentException(HERE) << "Error the lower bound a of a PiecewiseLinearDistribution distribution must be less than its upper bound b, here a=" << a << " b=" << b;
+  setDimension( 1 );
+  computeRange();
+}
+
+/* Comparison operator */
+Bool PiecewiseLinearDistribution::operator ==(const PiecewiseLinearDistribution & other) const
+{
+  if (this == &other) return true;
+  return (a_ == other.a_) && (b_ == other.b_);
+}
+
+Bool PiecewiseLinearDistribution::equals(const DistributionImplementation & other) const
+{
+  const PiecewiseLinearDistribution* p_other = dynamic_cast<const PiecewiseLinearDistribution*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String PiecewiseLinearDistribution::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << PiecewiseLinearDistribution::GetClassName()
+      << " name=" << getName()
+      << " dimension=" << getDimension()
+      << " a=" << a_
+      << " b=" << b_;
+  return oss;
+}
+
+String PiecewiseLinearDistribution::__str__(const String & ) const
+{
+  OSS oss(false);
+  oss << getClassName() << "(a = " << a_ << ", b = " << b_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+PiecewiseLinearDistribution * PiecewiseLinearDistribution::clone() const
+{
+  return new PiecewiseLinearDistribution(*this);
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void PiecewiseLinearDistribution::computeRange()
+{
+  setRange(Interval(a_, b_));
+}
+
+
+/* Get one realization of the distribution */
+Point PiecewiseLinearDistribution::getRealization() const
+{
+  return Point(1, a_ + (b_ - a_) * RandomGenerator::Generate());
```

Comment:
weird, thats the uniform code

Reply:
@jschueller I am currently removing the PiecewiseLinearDistribution class from this PR as it will not be ready for the RC. The other classes are ready for a review.

Reply:
got it